### PR TITLE
IGAPP-661: Move jest config

### PR DESCRIPTION
**Pull request by Taggotty**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

File: web-ts/jest.config.ts [hunk 1]
```diff
@@ -24,5 +27,10 @@ module.exports = {
       }
     ]
   ],
-  snapshotSerializers: ['enzyme-to-json/serializer']
+  snapshotSerializers: ['enzyme-to-json/serializer'],
+  globals: {
+    __BUILD_CONFIG__: webIntegreatTestCmsBuildConfig
+  }
```
Comment (Taggotty) [inline web-ts/jest.config.ts:33]: as `__BUILD_CONFIG__` is a const we cannot (at least not without typescript error)  override it in the `jest.setup.ts` another solution is to override it right here. This will not throw any error (as jest probably does the parsing) and works perfectly fine (which means I tested it 👍🏼 )

**Review by maxammann — APPROVED** [review]

**Comment by maxammann** [comment]
I think the package.json still contains a wrong config name as it ends now with .ts

File: web-ts/jest.setup.ts [hunk 2]
```diff
@@ -28,6 +24,7 @@ function walkDir(dir: string, callback: (dir: string) => void) {
 // modules automatically as soon as they are found
 const mocksPath = 'src/__mocks__/'
 const jsPath = '.js' // This only unmocks .js files not .json for example
+
 walkDir(mocksPath, name => {
   if (name.endsWith(jsPath)) {
     jest.unmock(name.substring(mocksPath.length, name.length - jsPath.length))
```
Comment (klinzo) [inline web-ts/jest.setup.ts:30]: This should be adjusted the same way as in the native project. Otherwise jest will automatically mock everything with a file in __mocks__

File: web-ts/src/__mocks__/redux-first-router-link.tsx [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import * as React from 'react'
```
Comment (klinzo) [inline web-ts/src/__mocks__/redux-first-router-link.tsx:1]: Delete, we won't use redux-first-router (-link) anymore.

**Review by klinzo — APPROVED** [review]

File: web-ts/src/components/__tests__/Failure.spec.tsx [hunk 4]
```diff
@@ -1,13 +1,14 @@
...
 import { Failure } from '../Failure'
+import { MemoryRouter } from 'react-router-dom'
 
 describe('Failure', () => {
   const mockTranslate = key => key
 
   it('should render a simple failure and match snapshot', () => {
     const errorMessage = 'error message'
-    const { getByText } = render(<Failure errorMessage={errorMessage} t={mockTranslate} />)
+    const { getByText } = render(<Failure errorMessage={errorMessage} t={mockTranslate} />, { wrapper: MemoryRouter })
```
Comment (klinzo) [inline web-ts/src/components/__tests__/Failure.spec.tsx:11]: Why is this needed?

File: web-ts/src/components/__tests__/FailureSwitcher.spec.tsx [hunk 5]
```diff
@@ -31,19 +30,19 @@ describe('FailureSwitcher', () => {
     it('should call render content not found component and create a Failure component', () => {
       const error = new NotFoundError({ type, id, language, city })
       const spy = jest.spyOn(FailureSwitcher, 'renderContentNotFoundComponent')
```
Comment (klinzo) [inline web-ts/src/components/__tests__/FailureSwitcher.spec.tsx:32]: I think we could remove this, can't we? The check seems quite unnecessary.

**Review by klinzo — APPROVED** [review]
